### PR TITLE
ui: add custom form for update template

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Adds a custom UI form for updating a template with more options such as details.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):
Screenshot from 2021-09-14 18-03-29

### CODE REVIEW

**Comment by sureshanaparti** [comment]
@blueorangutan ui

File: ui/src/views/image/UpdateTemplate.vue [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+        for (const key in values) {
+          if (!this.isValidValueForKey(values, key)) continue
+          if (detailsField.includes(key)) {
+            params['details[0].' + key] = values[key]
+            continue
+          }
+          params[key] = values[key]
+        }
+        api('updateTemplate', params).then(json => {
+          this.$message.success(`${this.$t('message.success.update.template')}: ${this.resource.name}`)
```
Comment (utchoang) [inline ui/src/views/image/UpdateTemplate.vue:364]: @shwstppr Can you add the translate this key.
image

**Comment by sureshanaparti** [comment]
Hi @shwstppr can you address the outstanding comments please. thanks.

**Comment by shwstppr** [comment]
@blueorangutan ui

**Review by utchoang — APPROVED** [review]
LGTM!

**Comment by sureshanaparti** [comment]
@shwstppr Checked with KVM & VMware templates. I could see the keyboard type in the API call from UI, but I'm re-open the update template form, the updated keyboard type is not shown. Please check.

File: ui/src/views/image/UpdateTemplate.vue [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+      if (this.loading) return
+      this.form.validateFields((err, values) => {
+        if (err) {
+          return
+        }
+        this.loading = true
+        const params = {
+          id: this.resource.id
+        }
+        const detailsField = ['rootDiskController', 'nicAdapter', 'keyboardType']
```
Comment (sureshanaparti) [inline ui/src/views/image/UpdateTemplate.vue:354]:
```suggestion
        const detailsField = ['rootDiskController', 'nicAdapter', 'keyboard']
```

May be, this is the issue with keyboard detail.

**Comment by shwstppr** [comment]
@sureshanaparti fixed the empty nicAdapter, keyboard behaviour. If no value is selected for these fields, detail will not be passed to the API and it will be removed for the template.
Added warning for changing template type to SYSTEM
Screenshot from 2022-01-24 12-26-48

**Comment by shwstppr** [comment]
@blueorangutan ui

**Comment by sureshanaparti** [comment]
> @sureshanaparti fixed the empty nicAdapter, keyboard behaviour. If no value is selected for these fields, detail will not be passed to the API and it will be removed for the template.
> Added warning for changing template type to SYSTEM
> Screenshot from 2022-01-24 12-26-48

ok @shwstppr , will check. thanks.

**Review by sureshanaparti — APPROVED** [review]
LGTM

**Comment by Pearl1594** [comment]
@shwstppr I believe we need to disable / remove the templatetype field from the form when a user with User role logs in -
image

**Comment by shwstppr** [comment]
@blueorangutan ui

**Comment by shwstppr** [comment]
@Pearl1594 can you please try again

**Comment by shwstppr** [comment]
@blueorangutan ui

**Review by Pearl1594 — APPROVED** [review]
Tested as root admin and user - works as expected. LGTM

**Pull request merged**